BridgeCareApp/VuejsApp/src/App.vue

```diff
@@ -0,0 +1,216 @@
+                    authorityPR: this.passwordResetPolicy
+                };
+            }
+            else {
+                this.applicationConfig = {
+                    clientID: 'c4132c6f-fbb2-489c-bdf5-dda2ce009af8',
+                    graphScopes: ['https://aratranstest.onmicrosoft.com/userStage/user_impersonation'],
+                    authority: this.signupSignInPolicy,
+                    authorityPR: this.passwordResetPolicy
+                };
+            }
+            this.clientApp = new Msal.UserAgentApplication(
+                this.applicationConfig.clientID, this.applicationConfig.authority,
+                (errorDesc, token, error, tokenType) => {
+                    //this.getSilentTokenForLogin1(errorDesc, token, error, tokenType);
+                    console.log(token);
+                },
+                {
+                    cacheLocation: 'localStorage',
+                    //redirectUri: redirectUri
+                }
+            );
+        }
+
+        data() {
+            return {
+                items: [
+                    { navigation: 'Home', icon: 'dashboard', name: 'Home' },
+                    { navigation: 'TODO', icon: 'question_answer', name: 'TODO' },
+                    { navigation: 'ReportData', icon: 'receipt', name: 'Report data'}
+                ],
+                drawer: true
+            }
+        }
+
+        routing(routeName: string) {
+            this.$router.push(routeName);
+        }
+
+        created() {
+            let user = this.clientApp.getUser();
+            if (!user) {
+                //this.login();
+            }
+            else {
+                //this.getSilentToken();
+                this.loginFailed = false;
+                this.updateUI();
+            }
+        }
+
+        login() {
+            this.loginFailed = true;
+                this.clientApp.loginPopup(this.applicationConfig.graphScopes).then(
+                    idToken => {
+                        this.loginFailed = false;
+                        this.updateUI();
+                    //this.getSilentToken();
+                },
+                (error) => {
+                    this.redirectOnErrors(error)
+                }
+            );
+        }
+
+        getSilentToken() {
+            this.clientApp.acquireTokenSilent(this.applicationConfig.graphScopes).then(
+                accessToken => {
+                    this.loginFailed = false;
+                    this.updateUI();
+                }, error => {
+                    this.clientApp.acquireTokenPopup(this.applicationConfig.graphScopes).then(accessToken => {
+                        this.loginFailed = false;
+                        this.updateUI();
+                    }, error => {
+                        this.logMessage("Error acquiring the token:\n" + error);
+                    });
+                }
+            )
+        }
+
+        redirectOnErrors(error: any) {
```

Comment:
Denote this with the [HACK] comment prefix and describe the hack details in the comment block.

Reply:
added the comment